docs/tutorials/sensormanagement/03_OptimisedSensorManagement.py

```diff
@@ -0,0 +1,631 @@
+                                associator=associator)
+
+# %%
+# For each time step, data is added to the metric manager on truths and tracks.
+# The metrics themselves can then be generated from the metric manager.
+
+metric_managerA.add_data(truths, tracksA)
+metric_managerB.add_data(truths, tracksB)
+metric_managerC.add_data(truths, tracksC)
+
+metricsA = metric_managerA.generate_metrics()
+metricsB = metric_managerB.generate_metrics()
+metricsC = metric_managerC.generate_metrics()
+
+# %%
+# OSPA metric
+# ^^^^^^^^^^^
+#
+# First we look at the OSPA metric. This is plotted over time for each sensor manager method.
+
+import matplotlib.pyplot as plt
+
+ospa_metricA = metricsA['OSPA distances']
+ospa_metricB = metricsB['OSPA distances']
+ospa_metricC = metricsC['OSPA distances']
+
+fig = plt.figure()
+ax = fig.add_subplot(1, 1, 1)
+ax.plot([i.timestamp for i in ospa_metricA.value],
+        [i.value for i in ospa_metricA.value],
+        label='BruteForceSensorManager')
+ax.plot([i.timestamp for i in ospa_metricB.value],
+        [i.value for i in ospa_metricB.value],
+        label='OptimizeBruteSensorManager')
+ax.plot([i.timestamp for i in ospa_metricC.value],
+        [i.value for i in ospa_metricC.value],
+        label='OptimizeBasinHoppingSensorManager')
+ax.set_ylabel("OSPA distance")
+ax.set_xlabel("Time")
+ax.legend()
+
+# %%
+# SIAP metrics
+# ^^^^^^^^^^^^
+#
+# Next we look at SIAP metrics. We are only interested in the positional accuracy (PA) and
+# velocity accuracy (VA). These metrics can be plotted to show how they change over time.
+
+fig, axes = plt.subplots(2)
+
+times = metric_managerA.list_timestamps()
+
+pa_metricA = metricsA['SIAP Position Accuracy at times']
+va_metricA = metricsA['SIAP Velocity Accuracy at times']
+
+pa_metricB = metricsB['SIAP Position Accuracy at times']
+va_metricB = metricsB['SIAP Velocity Accuracy at times']
+
+pa_metricC = metricsC['SIAP Position Accuracy at times']
+va_metricC = metricsC['SIAP Velocity Accuracy at times']
+
+axes[0].set(title='Positional Accuracy', xlabel='Time', ylabel='PA')
+axes[0].plot(times, [metric.value for metric in pa_metricA.value],
+             label='BruteForceSensorManager')
+axes[0].plot(times, [metric.value for metric in pa_metricB.value],
+             label='OptimizeBruteSensorManager')
+axes[0].plot(times, [metric.value for metric in pa_metricC.value],
+             label='OptimizeBasinHoppingSensorManager')
+axes[0].legend()
+
+axes[1].set(title='Velocity Accuracy', xlabel='Time', ylabel='VA')
+axes[1].plot(times, [metric.value for metric in va_metricA.value],
+             label='BruteForceSensorManager')
+axes[1].plot(times, [metric.value for metric in va_metricB.value],
+             label='OptimizeBruteSensorManager')
+axes[1].plot(times, [metric.value for metric in va_metricC.value],
+             label='OptimizeBasinHoppingSensorManager')
+axes[1].legend()
+
+# %%
+# Uncertainty metric
```

Comment:
I feel like this plot could be made more impactful by reducing the covariance in the prior states, or by making the sensor models more noisy. This is something that's subjective though, so feel free to ignore me!

Reply:
Is this for the uncertainty metric graph? Good point, I may also remove the OSPA one as it doesn't show anything useful

Reply:
Yes, the uncertainty metric plot.